streaming/streaming.go

```diff
@@ -0,0 +1,258 @@
+	dsCfg := streamingCfg.DatastreamCfg
+	if dsCfg.StreamLocation == "" {
+		return fmt.Errorf("please specify DatastreamCfg.StreamLocation in the streaming config")
+	}
+	srcCfg := dsCfg.SourceConnectionConfig
+	if srcCfg.Name == "" || srcCfg.Location == "" {
+		return fmt.Errorf("please specify Name and Location under DatastreamCfg.SourceConnectionConfig in the streaming config")
+	}
+	dstCfg := dsCfg.DestinationConnectionConfig
+	if dstCfg.Name == "" || dstCfg.Location == "" {
+		return fmt.Errorf("please specify Name and Location under DatastreamCfg.DestinationConnectionConfig in the streaming config")
+	}
+
+	dfCfg := streamingCfg.DataflowCfg
+	if dfCfg.Location == "" {
+		return fmt.Errorf("please specify the Location under DataflowCfg in the streaming config")
+	}
+
+	// If both ID and Display name are empty, generate a new one for both.
+	// If either is present, assign it to the other one.
+	if dsCfg.StreamId == "" && dsCfg.StreamDisplayName == "" {
+		// TODO: Update names to have more info like dbname.
+		streamId, err := utils.GenerateName("hb-stream-" + dbName)
+		if err != nil {
+			return fmt.Errorf("error generating stream name: %v", err)
+		}
+		streamingCfg.DatastreamCfg.StreamId = streamId
+		streamingCfg.DatastreamCfg.StreamDisplayName = streamId
+	} else if dsCfg.StreamId == "" {
+		streamingCfg.DatastreamCfg.StreamId = streamingCfg.DatastreamCfg.StreamDisplayName
+	} else if dsCfg.StreamDisplayName == "" {
+		streamingCfg.DatastreamCfg.StreamDisplayName = streamingCfg.DatastreamCfg.StreamId
+	}
+
+	if dfCfg.JobName == "" {
+		// Update names to have more info like dbname.
+		jobName, err := utils.GenerateName("hb-dataflow-" + dbName)
+		if err != nil {
+			return fmt.Errorf("error generating stream name: %v", err)
+		}
+		streamingCfg.DataflowCfg.JobName = jobName
+	}
+	return nil
+}
+
+// ReadStreamingConfig reads the file and unmarshalls it into the StreamingCfg struct.
+func ReadStreamingConfig(file, dbName string) (StreamingCfg, error) {
+	streamingCfg := StreamingCfg{}
+	cfgFile, err := ioutil.ReadFile(file)
+	if err != nil {
+		return streamingCfg, fmt.Errorf("can't read streaming config file due to: %v", err)
+	}
+	err = json.Unmarshal(cfgFile, &streamingCfg)
+	if err != nil {
+		return streamingCfg, fmt.Errorf("unable to unmarshall json due to: %v", err)
+	}
+	err = VerifyAndUpdateCfg(&streamingCfg, dbName)
+	if err != nil {
+		return streamingCfg, fmt.Errorf("streaming config is incomplete: %v", err)
+	}
+	return streamingCfg, nil
+}
+
+// LaunchStream populates the parameters from the streaming config and triggers a stream on Cloud Datastream.
+func LaunchStream(ctx context.Context, dbName, projectID string, datastreamCfg DatastreamCfg) error {
+	fmt.Println("Launching stream...")
+	dsClient, err := datastream.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("datastream client can not be created: %v", err)
+	}
+	defer dsClient.Close()
+	fmt.Println("Created client...")
```

Comment:
add details of client created in the logs

Reply:
The datastream client just takes the context as an input when instantiated. So there is not any detail to put in the logs.